### ISSUE TITLE
Add support for overlay_getLogs and overlay_callConstructor

### ISSUE THREAD

**Issue by joshstevens19** [paradigmxyz/reth#7503]
### Describe the feature

We just seen erigon just merge a PR to allow you with a single RPC request to modify the contract injecting your own events and get the new logs back.. more information on the issue: https://github.com/ledgerwatch/erigon/pull/9438

This is making people who switched from erigon to reth going back purely for how strong this feature is. 

Would be cool to see it here!

**Comment by sambacha** [comment]
sorry first this needs to be done https://github.com/paradigmxyz/reth/issues/6752

get back in line 👿

**Comment by andy-thomason** [comment]
Forgive me if I'm mistaken, but can this be done already with the Geth state override set in eth_call which is also supported by Reth?

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/api/call.rs

**Comment by gakonst** [comment]
Yes that works on a block by block basis, but doesn't have a backfill option for re-execution of N blocks in 1 RPC call w/ overridden bytecode to get new logs out!

**Comment by crebsy** [comment]
hey, glad to see this issue here!

Here are some integration tests for a few contracts deployed on ethereum mainnet:
https://github.com/ledgerwatch/erigon/blob/devel/cmd/rpcdaemon/postman/Overlay_Testing.json
Anyone who plans implementing the two new rpc methods, can run those tests against an archive node with postman until they're all :green_circle: 

There's also a minimal README for overlays: https://github.com/ledgerwatch/erigon/blob/devel/turbo/jsonrpc/overlay/README.md
but the tests are probably more fun to play with :smile:
